src/main/java/uk/gov/pay/adminusers/service/SelfSignupInviteCompleter.java

```diff
@@ -52,8 +53,8 @@ public InviteCompleteResponse complete(InviteEntity inviteEntity) {
         if (inviteEntity.isServiceType()) {
             UserEntity userEntity = inviteEntity.mapToUserEntity();
             ServiceEntity serviceEntity = ServiceEntity.from(Service.from());
-            if (!data.getGatewayAccountIds().isEmpty()) {
-                serviceEntity.addGatewayAccountIds(data.getGatewayAccountIds().toArray(new String[0]));
+            if (completeInviteRequest != null && !completeInviteRequest.getGatewayAccountIds().isEmpty()) {
+                serviceEntity.addGatewayAccountIds(completeInviteRequest.getGatewayAccountIds().toArray(new String[0]));
```

Comment:
Having now read through the code a bit more, I wonder if it's OK to just remove the request body concept entirely and just remove this behaviour, appreciate that would involve updating a lot of tests so maybe for a separate change

Reply:
Ah cool, I'll do it in a separate PR then. Would be good to get this point of confusion removed while its in our heads.